frontend/app/src/util/AppNavigation.ts

```diff
@@ -39,32 +39,17 @@ export type PageUrlUpdateCallback = (
 ) => void
 export type PageNotFoundCallback = (pageName?: string) => void
 
-export class AppNavigation {
-  readonly hostCommunicationMgr: HostCommunicationManager
-
-  readonly metricsMgr: SegmentMetricsManager
-
-  readonly onUpdatePageUrl: PageUrlUpdateCallback
-
-  readonly onPageNotFound: PageNotFoundCallback
-
-  private appPages: IAppPage[]
+export class V1Strategy {
+  appPages: IAppPage[]
 
-  private currentPageScriptHash: string | null
+  currentPageScriptHash: string | null
 
-  private hideSidebarNav: boolean | null
+  hideSidebarNav: boolean | null
 
-  constructor(
-    hostCommunicationMgr: HostCommunicationManager,
-    metricsMgr: SegmentMetricsManager,
-    onUpdatePageUrl: PageUrlUpdateCallback,
-    onPageNotFound: PageNotFoundCallback
-  ) {
-    this.hostCommunicationMgr = hostCommunicationMgr
-    this.metricsMgr = metricsMgr
-    this.onUpdatePageUrl = onUpdatePageUrl
-    this.onPageNotFound = onPageNotFound
+  parent: AppNavigation
```

Comment:
nit: idk how I feel about the `parent` name here vs just calling this something like `appNav`. No strong opinion here though